packages/docusaurus-theme-classic/src/__tests__/validateThemeConfig.test.js

```diff
@@ -345,6 +345,68 @@ describe('themeConfig', () => {
     });
   });
 
+  test('should allow simple links in footer', () => {
+    const partialConfig = {
+      footer: {
+        links: [
+          {
+            label: 'Privacy',
+            href: 'https://opensource.facebook.com/legal/privacy/',
+          },
+          {
+            label: 'Terms',
+            href: 'https://opensource.facebook.com/legal/terms/',
+          },
+          {
+            label: 'Data Policy',
+            href: 'https://opensource.facebook.com/legal/data-policy/',
+          },
+          {
+            label: 'Cookie Policy',
+            href: 'https://opensource.facebook.com/legal/cookie-policy/',
+          },
+        ],
+      },
+    };
+    const normalizedConfig = testValidateThemeConfig(partialConfig);
+
+    expect(normalizedConfig).toEqual({
+      ...normalizedConfig,
+      footer: {
+        ...normalizedConfig.footer,
+        ...partialConfig.footer,
+      },
+    });
+  });
+
+  test('should reject mix of simple and multi-column links in footer', () => {
+    const partialConfig = {
+      footer: {
+        links: [
+          {
+            title: 'Learn',
+            items: [
+              {
+                label: 'Introduction',
+                to: 'docs',
+              },
+            ],
+          },
+          {
+            label: 'Privacy',
+            href: 'https://opensource.facebook.com/legal/privacy/',
+          },
+        ],
+      },
+    };
+
+    expect(() =>
+      testValidateThemeConfig(partialConfig),
+    ).toThrowErrorMatchingInlineSnapshot(
+      `"\\"footer.links\\" does not match any of the allowed types"`,
```

Comment:
The message doesn't seem so friendly to me. I know custom messages are very hard in Joi, but could you try to improve that to "the footer must either be simple or multi-column, but not a mix, see https://docusaurus.io/docs/api/themes/configuration#footer-links", or something like that?

Reply:
Completely agree! Should be fixed now.